tfhe/src/core_crypto/fft_impl/fft64/crypto/wop_pbs/mod.rs

```diff
@@ -551,7 +550,11 @@ pub fn cmux_tree_memory_optimized<Scalar: UnsignedTorus + CastInto<usize>>(
                         ciphertext_modulus,
                     );
 
-                    if j != nb_layer - 1 {
+                    if j == nb_layer - 1 {
+                        let mut output = output_glwe.as_mut_view();
```

Comment:
https://rust-lang.github.io/rust-clippy/master/index.html#/if_not_else

Reply:
could we have a check on the if ! pattern and update those that may make more sense ? :)

here for example j < nb_layer - 1 should have been used instead of j != nb_layer - 1